src/create-context.js

```diff
@@ -52,7 +52,9 @@ export function createContext(defaultValue, contextId) {
 					subs.push(c);
 					let old = c.componentWillUnmount;
 					c.componentWillUnmount = () => {
-						subs.splice(subs.indexOf(c), 1);
+						// This is a hot path so we want to be careful about how we're removing items from the subscriber array to keep this fast.
+						// This is faster than splicing an item out in testing.
+						subs[subs.indexOf(c)] = subs.pop();
```

Comment:
I think this doesn't work if `c` is the last element in the array, since it'll reassign the last element back to it's original index. I think we have to do this in two lines.

Reply:
oh right, I missed that in my review. Also what happens if there is only one entry in the array?

Reply:
We could consider using `Set` for this if the Array is a pain, or switching to a sparse array (would require a guard on L45).